custom/ilsgateway/tasks.py

```diff
@@ -364,3 +368,67 @@ def third_soh_task():
     last_month_last_day, fifth_business_day = get_last_and_nth_business_day(now, 5)
     if now.day == fifth_business_day.day:
         send_for_all_domains(last_month_last_day, SOHReminder)
+
+
+def recalculate_on_group_change(location, last_run):
+    OrganizationSummary.objects.filter(location_id=location.get_id).delete()
+    process_facility_warehouse_data(location, default_start_date(), last_run.end)
+
+    for parent in location.sql_location.get_ancestors(ascending=True):
```

Comment:
can you explain in a little more detail why reprocessing this for all ancestors of the current location is all we need to do? (i'm not saying this is wrong, i may just not be understanding how location reporting groups work)